### PR TITLE
Feat/lesson list item numbered

### LINKED ISSUE #1419: I want lessons to be numbered on the lesson listing page
In figma, each lesson is numbered. There is currently no filtering happening on the front end so no considerations needed there.

https://www.figma.com/file/hQoMOEvJQlDJZ0n2ehftVT/Beta?node-id=1481-456677&t=Mc6eoz34Nqr40L11-4

--- 
**Acceptance Criteria:**
- [ ] Lessons on a lesson listing page should be numbered corresponding to where they appear in the list of displayed lessons 
    - Not sure if it will be a trip up, but worth checking in KS4 maths, tiered as there are shared lesson, though I don't think any funny business will happen there
- [ ] Number expired lessons as if they were in the sequence

### PR DESCRIPTION
**Pull request by JBR90**
## Description

- Adds numbers to lesson list items

## Issue(s)

Fixes #1419 

## How to test

1. Go to lesson list page and you should see numbers, search lesson list items should have no numbers
2. Click on _______
3. You should see _______

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner

### CODE REVIEW

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
Pagination seems to break things! 
https://deploy-preview-1605--oak-web-application.netlify.thenational.academy/beta/teachers/programmes/english-primary-ks2/units/how-to-train-your-dragon-mixed-outcomes-50a0/lessons?page=1

Screenshot 2023-05-15 at 10 54 43

**Review by jack-skerrett-oak — APPROVED** [review]

File: src/node-lib/curriculum-api/index.ts [hunk 1]
```diff
@@ -399,13 +399,24 @@ const curriculumApi = {
       programmesUnavailable,
     } = transformMVCase(res);
 
+    let unavailableProgrammesWithoutAvailable;
+
+    if (programmesAvailable && programmesUnavailable) {
```
Comment (benprotheroe) [inline src/node-lib/curriculum-api/index.ts:404]: will this overide the refactoring from your other ticket?

**Review by benprotheroe — APPROVED** [review]

File: src/node-lib/curriculum-api/index.ts [hunk 2]
```diff
@@ -399,13 +399,24 @@ const curriculumApi = {
       programmesUnavailable,
     } = transformMVCase(res);
 
+    let unavailableProgrammesWithoutAvailable;
```
Comment (k-huggs) [inline src/node-lib/curriculum-api/index.ts:402]: Is this the same code that you worked on in your other PR?

Comment (JBR90) [reply src/node-lib/curriculum-api/index.ts]: Pushed to the wrong branch, thats gone now

**Pull request merged**